src/libstd/io/mod.rs

```diff
@@ -505,24 +859,69 @@ fn read_until<R: BufRead + ?Sized>(r: &mut R, delim: u8, buf: &mut Vec<u8>)
     }
 }
 
-/// A `BufRead` is a type of reader which has some form of internal buffering to
-/// allow certain kinds of reading operations to be more optimized than others.
+/// A `BufRead` is a type of `Read`er which has an internal buffer, allowing it
+/// to perform extra ways of reading.
+///
+/// For example, reading line-by-line requires using a buffer, so if you want
```

Comment:
s/requires using a buffer/is inefficient without a buffer/

You can technically read line-by-line by just reading one byte at a time, but there's no way that anyone actually wants to do that.